DiamondLightSource/python-artemis
# Mxgda 3734: Create nexus file

See https://jira.diamond.ac.uk/browse/MXGDA-3734

Note that this includes https://github.com/DiamondLightSource/python-artemis/pull/51 so review that first.

This also requires new dependencies so you will need to reinstall them.

To test:
* Confirm unit tests pass
* Copy some existing `*_00001.h5` and `*_meta.h5` data from e.g. `/dls/i03/data/2022/cm31105-1/xraycentring/TestProteinaseK/protk_1` to a local folder
* Run Artemis and send it parameters with:
```
"detector_params": {"acquisition_id": "test",
                     "current_energy": 100,
                     "detector_distance": 100.0,
                     "directory": path_to_local_data_folder,
                     "exposure_time": 0.1,
                     "num_images": 10,
                     "omega_increment": 0.1,
                     "omega_start": 0.0,
                     "prefix": prefix_of_file_eg_protk_1_1,
                     "use_roi_mode": false},
```
* Confirm a nexus file is written in the local folder
* Compare with the nexus file from the original folder and see that they follow a broadly similar structure
